lib/index.d.ts

```diff
@@ -295,3 +356,15 @@ export class Component<
    */
   getContext<ContextKey extends keyof ContextRegistryT>(contextName: ContextKey): ContextRegistryT[ContextKey];
 }
+
+/**
+ * Panel component that only accepts 3 generic types
+ */
+export class Panel<ParamT = unknown, StateT = unknown, ContextRegistryT = unknown> extends Component<
```

Comment:
`Panel` is a bit of an odd name for it, how about something explicit like `ParamComponent`